File: MamsLeagueApi/MamsLeagueApi/Models/Dtos/FixtureDto.cs
using System.ComponentModel.DataAnnotations;

namespace MamsLeagueApi.Models.Dtos
{
    public class FixtureDto
    {
        public Guid Id { get; set; }
        [Required]
        public Guid SchedulePlayId { get; set; }
        [Required]
        public Guid GroundId { get; set; }
        [Required]
        public Guid PlayTimeId { get; set; }
        [Required]
        public Guid AwayTeamId { get; set; }
        public int? AwayTeamScore { get; set; }
        [Required]
        public Guid HomeTeamId { get; set; }
        public int? HomeTeamScore { get; set; }
        public virtual TeamDto? AwayTeamDto { get; set; }
        public virtual GroundDto? GroundDto { get; set; }
        //[InverseProperty("HomeTeamId")]
        public virtual TeamDto? HomeTeamDto { get; set; }
        public virtual PlayTimeDto? PlayTimeDto { get; set; }
        public virtual SchedulePlayDto? SchedulePlayDto { get; set; }
    }
}

File: mamsleagueweb/src/app/model/dtos/fixture_dto.ts
import { GroundDto } from "./ground_dto";
import { PlayTimeDto } from "./play_time_dto";
import { SchedulePlayDto } from "./schedule_play_dto";
import { TeamDto } from "./team_dto";

export interface FixtureDto {
    id: string;
    schedulePlayId: string;
    groundId: string;
    playTimeId: string;
    awayTeamId: string;
    awayTeamScore?: number | undefined;
    homeTeamId: string;
    homeTeamScore?: number | undefined;
    awayTeamDto?: TeamDto | undefined;
    groundDto?: GroundDto | undefined;
    homeTeamDto?: TeamDto | undefined;
    playTimeDto?: PlayTimeDto | undefined;
    schedulePlayDto?: SchedulePlayDto | undefined;
}
